modules/platform/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/requestcontroller/filter/IdempotentRequestHandler.java

```diff
@@ -0,0 +1,168 @@
+ * progress.
+ * </ul>
+ *
+ * @since 11.5
+ */
+public class IdempotentRequestHandler {
+
+    private static final Logger log = LogManager.getLogger(IdempotentRequestHandler.class);
+
+    /** @since 11.5 */
+    public static final String HEADER_KEY = "Idempotency-Key";
+
+    /** @since 11.5 */
+    public static final String STORE_PROPERTY = "org.nuxeo.request.controller.keyvaluestore.name";
+
+    protected static final String DEFAULT_STORE = "idempotentrequest";
+
+    /** 1 day */
+    protected static final int DEFAULT_TTL_SECONDS = 86400;
+
+    /** @since 11.5 */
+    public static final String TTL_SECONDS_PROPERTY = "org.nuxeo.request.controller.ttl.seconds";
+
+    /** @since 11.5 */
+    public static final String INPROGRESS_MARKER = "IDEMPOTENCY_INPROGRESS_MARKER";
+
+    /** @since 11.5 */
+    public static final String STATUS_SUFFIX = "_status";
+
+    // safe methods according to RFC 7231 4.2.1
+    protected static final Set<String> SAFE_METHODS = new HashSet<>(
+            Arrays.asList(HttpGet.METHOD_NAME, HttpHead.METHOD_NAME, HttpOptions.METHOD_NAME, HttpTrace.METHOD_NAME));
+
+    // idempotent methods according to RFC 7231 4.2.2
+    protected static final Set<String> IDEMPOTENT_METHODS = new HashSet<>(
+            Arrays.asList(HttpPut.METHOD_NAME, HttpDelete.METHOD_NAME));
+
+    protected int getTTLSeconds() {
+        ConfigurationService cs = Framework.getService(ConfigurationService.class);
+        if (cs != null) {
+            return cs.getInteger(TTL_SECONDS_PROPERTY, DEFAULT_TTL_SECONDS);
+        }
+        return DEFAULT_TTL_SECONDS;
+    }
+
+    protected String getStoreName() {
+        ConfigurationService cs = Framework.getService(ConfigurationService.class);
+        if (cs != null) {
+            return cs.getString(STORE_PROPERTY, DEFAULT_STORE);
+        }
+        return DEFAULT_STORE;
+    }
+
+    protected boolean shouldIgnore(String method) {
+        return SAFE_METHODS.contains(method) || IDEMPOTENT_METHODS.contains(method);
+    }
+
+    public void handle(HttpServletRequest request, HttpServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        String method = request.getMethod();
+        log.debug("Method: " + method);
+        if (shouldIgnore(method)) {
+            log.debug("Request ignored");
+            chain.doFilter(request, response);
+            return;
+        }
+        String key = request.getHeader(IdempotentRequestHandler.HEADER_KEY);
+        if (key == null) {
+            log.debug("No idempotent request key");
+            chain.doFilter(request, response);
+            return;
+        }
+        log.debug("Idempotent request key: " + key);
+        KeyValueService kvs = Framework.getService(KeyValueService.class);
+        if (kvs == null) {
+            log.debug("KeyValueStore not present");
+            chain.doFilter(request, response);
```

Comment:
You have many `if (...) { chain.doFilter(request, response); return; }` so maybe let the caller do the chain call if this method returns a boolean `false`.

Reply:
I did not change this, but since in the end there is a dedicated filter maybe it's alright as is?